src/templates/search.liquid

```diff
@@ -2,11 +2,8 @@
 
   <h1>
     {% if search.performed %}
-      {% if search.results_count == 0 %}
-        {{ 'general.search.no_results_html' | t: terms: search.terms }}
-      {% else %}
-        {{ 'general.search.results_for_html' | t: terms: search.terms }}
-      {% endif %}
+      <span class="visually-hidden">{{ 'general.search.heading' | t }}:</span>
```

Comment:
My bad, I misunderstood what you were trying to do here. You're going to have to revert to what you had and add:

```liquid
{{ 'general.search.heading' | t: count: search.results_count }}
```